member-service/src/main/java/com/hedvig/memberservice/services/metrics/MetricsCollectorImpl.kt

```
@@ -0,0 +1,76 @@
+    }
+
+    override fun registerSwedishBankIdSignInitiation() {
+        registerEvent("bank_id.se.sign.init")
+    }
+
+    override fun registerSwedishBankIdSignCompletion(result: ExecutionResult) {
+        registerFailableEvent("bank_id.se.sign.complete", result)
+    }
+
+    override fun registerSwedishBankIdLoginInitiation() {
+        registerEvent("bank_id.se.login.init")
+    }
+
+    override fun registerSwedishBankIdLoginCompletion(result: ExecutionResult) {
+        registerFailableEvent("bank_id.se.login.complete", result)
+    }
+
+    override fun registerNorwegianBankIdSignInitiation() {
+        registerEvent("bank_id.no.sign.init")
+    }
+
+    override fun registerNorwegianBankIdSignCompletion(result: ExecutionResult) {
+        registerFailableEvent("bank_id.no.sign.complete", result)
+    }
+
+    override fun registerNorwegianBankIdLoginInitiation() {
+        registerEvent("bank_id.no.login.init")
+    }
+
+    override fun registerNorwegianBankIdLoginCompletion(result: ExecutionResult) {
+        registerFailableEvent("bank_id.no.login.complete", result)
+    }
+
+    private fun registerFailableEvent(title: String, result: ExecutionResult) {
+        logger.info("Sending ${result} event \"$title\" to DogStatsD")
+        val event = Event.builder()
+            .withAlertType(
+                when (result) {
+                    ExecutionResult.Success -> Event.AlertType.INFO
+                    ExecutionResult.Error -> Event.AlertType.ERROR
+                }
+            )
+            .withTitle(title)
+            .build()
+        dogStatsDClient.recordEvent(
+            event,
+            when (result) {
+                ExecutionResult.Success -> "status:success"
+                ExecutionResult.Error -> "status:error"
+            }
+        )
+    }
+
+    private fun registerEvent(title: String) {
+        logger.info("Sending event \"$title\" to DogStatsD")
```

If we worry even a little bit about exceptions in the DataDog-internals, we could wrap all this in `runCatching` as we're kind of fine with discarding the errors 🙂